### DISCUSSION TITLE
Code for PicoCNC

### DISCUSSION THREAD

**Discussion by peterashuk** [phil-barrett/PicoCNC#7] [General]
Hi, I've built up my PicoCNC V1.51 board and loaded firmware created from grblHAL webbuilder v0.9d
using:
RP2040
PicoCNC
Compatibility grblHAL       
The board connects but jog doesn't work as presumably the estop is active.  Nothing gets hot and I can't see any soldering issues.
Does anyone have any advice ?
Also, I assume that I have to tell the pico to use shift registers, does anyone know how to do this is the selection?

**Comment by phil-barrett** [comment]
You are all set to go.  grblHAL knows about the shift registers. The [user manual](https://github.com/phil-barrett/PicoCNC/blob/main/PicoCNC%20User%20Manual.pdf) describes what you need to do. 

Set $14 to 70 ($70=14) which will set Halt, Cycle start and Feed hold to normally open and you should be able to jog. I recommend ioSender for this as it describes the Grbl Settings and makes it easy to set them. You will need to power the board with 12V to actually use the inputs (limits, probe, controls).  Be careful if you jumper the 2 12V sections - make sure the jumpers are in the vertical positions.

**Reply by peterashuk** [reply]
I am using using ioSender and it now It now jogs ! Thanks for your prompt help.  I reflashed to compatiblity:10 - 

[VER:1.1f.20230825:]
[OPT:VNMSL,35,1024,3,0]
[AXS:3:XYZ]
[NEWOPT:ENUMS,RT+,SED,RTC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[FREE MEMORY:226K]
[DRIVER:RP2040]
[DRIVER VERSION:230820]
[DRIVER OPTIONS:SDK_1.5.0]
[BOARD:https://github.com/phil-barrett/PicoCNC]
[COMPATIBILITY LEVEL:10]
[AUX IO:4,8,0,1]

I will report back in due course.   Peter

**Comment by peterashuk** [comment]
Just to let you know that the PicoCNC board runs a stepper motor using a DM556 driver module !  Good progress.
Can I ask how you intended the 12v PSU to be provided ?  Can I use a buck converter(ss) to derive these from the main stepper motor PSU ?
Also, I am using compatibilty:10, should I re-flash to the default ?

**Reply by phil-barrett** [reply]
I would definitely go back to the default compat level. Level 10 is a throwback to when a lot of senders were not grblHAL aware. Not the case anymore.

12v from a buck converter would be great.

**Comment by peterashuk** [comment]
Hi Phil, Thanks for an excellent project, I've got the single 12v working using the jumpers.  Now most of the sensors have come alive.   I look forwards to more posts as more people get this product.  

![picocnc_12v](https://github.com/phil-barrett/PicoCNC/assets/2029387/582f522d-2dd1-464b-a1aa-8610a9926bd1)

**Reply by cpapplefamily** [reply]
Am I missing the mention to these jumpers in the documentation? I was having trouble getting things working until I seen this post

**Reply by phil-barrett** [reply]
It is called out in the User Manual in the "Powering the PicoCNC" section. Perhaps I should make it a little clearer.   Note that V1.52 of the board is slightly different.  I moved the two pin headers apart to make it impossible to jumper them incorrectly.

**Reply by cpapplefamily** [reply]
Ok I see the manual Now. My issue was I simply looked at the Git Rendered manual that only shows the first few pages. After downloading I found the missing information.

**Comment by phil-barrett** [comment]
Hmmm, that tells me I need to come up with a better way for people to get the manual.

**Reply by cpapplefamily** [reply]
Possibly the images with components installed should have the jumpers installed?

**Reply by phil-barrett** [reply]
Yeah, I knew this was going to be an issue - flexibility has it's cost.

**Reply by cpapplefamily** [reply]
Is there a possibility to put a 3pin header here and screen print 12vdc internal/  external. 

Also do you need to break the 0vdc line?

**Comment by phil-barrett** [comment]
Yes, you need to break the ground plane for true isolation.

**Reply by cpapplefamily** [reply]
Then maybe look at the 3 pin header idea to imply a choice like the 5v/3v UART and 5v 12v relay jumpers

Great work anyhow. I have my project functional. I'll share more when it all cleaned up.

**Reply by phil-barrett** [reply]
I've been through this a fair amount. The problem is that it is tied to use of the lower 12V screw terminals. It is indeed a choice but one choice requires additional action. Selecting isolated 12V with the jumper does nothing if a PSU is not hooked up. I dunno, none of the options are great. sigh...

Glad you are up and running!  Looking forward to seeing your machine!
